### PR TITLE
fix: [CI-6647]: Fix npe for null build type

### PR DESCRIPTION
**Pull request by aman-harness**
## Checklist
- [ ] I've documented the changes in the PR description.
- [ ] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- Immutable delegate `trigger publish-delegate`
</details>

<details>
  <summary>PR Check triggers</summary>

You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ti0, ti1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
    - CodeFormat: `trigger codeformat`
    - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- File-Permission-Check: `trigger checkpermission`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- PMD: `trigger pmd`
- Copyright Check: `trigger copyrightcheck`
- Feature Name Check: `trigger featurenamecheck`
- TI-bootstrap: `trigger ti0`
- TI-bootstrap1: `trigger ti1`
- TI-bootstrap2: `trigger ti2`
- TI-bootstrap3: `trigger ti3`
- TI-bootstrap4: `trigger ti4`
- FunctionalTest1: `trigger ft1`
- FunctionalTest2: `trigger ft2`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- Trigger all Checks: `trigger smartchecks`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/42377)

### CODE REVIEW

**Review by smjt-h — APPROVED** [review]
> Reviewed 1 of 6 files at r1, 5 of 5 files at r2, 3 of 3 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/42377)* status: :shipit: complete! ✓

File: 310-iacm-manager/build/container/scripts/replace_configs.sh [hunk 1]
```diff
@@ -220,6 +220,10 @@ if [[ "" != "$GRPC_SERVER_PORT" ]]; then
   export GRPC_SERVER_PORT; yq -i '.pmsSdkGrpcServerConfig.connectors[0].port=env(GRPC_SERVER_PORT)' $CONFIG_FILE
 fi
 
+if [[ "" != "$OVERRIDE_EXEC_LIMIT_FOR_ACCOUNT" ]]; then
+  export OVERRIDE_EXEC_LIMIT_FOR_ACCOUNT; yq -i '.ciExecutionServiceConfig.executionLimits.overrideConfig[0]=env(GRPC_SERVER_PORT)' $OVERRIDE_EXEC_LIMIT_FOR_ACCOUNT
```
Comment (shubham149) [inline 310-iacm-manager/build/container/scripts/replace_configs.sh:224]: GRPC_SERVER_PORT should not be a part of this line. Same for other places

File: dockerization/ci-manager/scripts/replace_configs.sh [hunk 2]
```diff
@@ -204,6 +204,10 @@ if [[ "" != "$TI_SERVICE_ENDPOINT" ]]; then
   export TI_SERVICE_ENDPOINT; yq -i '.tiServiceConfig.baseUrl=env(TI_SERVICE_ENDPOINT)' $CONFIG_FILE
 fi
 
+if [[ "" != "$OVERRIDE_EXEC_LIMIT_FOR_ACCOUNT" ]]; then
```
Comment (harshjain12) [inline dockerization/ci-manager/scripts/replace_configs.sh:207]: Also replace config  in /Users/harshjain/Development/harness-core/332-ci-manager/build/container/scripts/replace_configs.sh
332-ci-manager/container/scripts/replace_configs.sh

File: 310-iacm-manager/build/container/scripts/replace_configs.sh [hunk 3]
```diff
@@ -220,6 +220,10 @@ if [[ "" != "$GRPC_SERVER_PORT" ]]; then
   export GRPC_SERVER_PORT; yq -i '.pmsSdkGrpcServerConfig.connectors[0].port=env(GRPC_SERVER_PORT)' $CONFIG_FILE
 fi
 
+if [[ "" != "$OVERRIDE_EXEC_LIMIT_FOR_ACCOUNT" ]]; then
```
Comment (harshjain12) [inline 310-iacm-manager/build/container/scripts/replace_configs.sh:223]: what about sto ?

**Review by shubham149 — APPROVED** [review]
> Reviewed 1 of 6 files at r1, 3 of 5 files at r2, 1 of 3 files at r3, 1 of 2 files at r4, 1 of 1 files at r5, 2 of 2 files at r6, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/42377)* status: :shipit: complete! ✓

**Comment by aman-harness** [comment]
trigger messagecheck

**Pull request merged**